Auto link treatment stage to indication

9sneha-n opened the pull request:
### :pushpin: References

-   **Issue:** Closes #8696xvx8w,  https://app.clickup.com/t/8696xvx8w?

### :memo: Implementation
1. Auto link treatment stage to indication
### :video_camera: Screenshots/Screen capture

https://github.com/user-attachments/assets/79628fbd-8b84-4004-b24b-81dd4c9057b9


https://github.com/user-attachments/assets/6892e4a3-e786-4a0e-b9d7-e0d5eec9972d


https://github.com/user-attachments/assets/4310f8a4-cdea-4b3c-a7a0-e48f77506846


https://github.com/user-attachments/assets/9ee94c2f-200b-404e-84b7-91e56da0f32f


https://github.com/user-attachments/assets/4e86f5af-da4c-42e5-b932-81de418037ff


https://github.com/user-attachments/assets/4fe48bf7-6ffa-4b4e-9111-e3ea446bde76


https://github.com/user-attachments/assets/22870946-6d1a-40e6-89a2-894606882f45



### :fire: Notes to the tester

ifoche commented:
Task linked: [CU-8696xvx8w Add selector of indicators in treatments and selector of treatments in indicators](https://app.clickup.com/t/8696xvx8w)

MatiasArriola reviewed (commented):
Thanks @9sneha-n 
I think that most of the logic added to the `useTreatmentIndicationLink` hook belongs to the domain layer. Maybe to a method in `Questionnaire`? what do you think?

9sneha-n commented:
> Thanks @9sneha-n I think that most of the logic added to the `useTreatmentIndicationLink` hook belongs to the domain layer. Maybe to a method in `Questionnaire`? what do you think?

@MatiasArriola The reason I have added this logic in the custom hook, is because it is a presentation logic that we convert the text fields to dropdown and populate the linked treatments. What do you think?

MatiasArriola commented:
My reasoning is that the code in the hook primarily transforms domain entities `Questionnaire`, `QuestionnaireSection`, `Question` based on specific rules (e.g. linking certain questions bidirectionally). 
`Question` is still a domain entity even though it contains information that will be used by the presentation layer to render it (what kind of question is it? is it disabled?).
The validation limiting a treatment to a maximum of 5 indications feels like a business rule rather than presentation logic.
Another way to evaluate this is to ask: If we switched to a different presentation layer, would we need to rewrite this logic?

(unrelated to this PR) Ideally the data layer shouldn't be directly referenced by the presentation layer, though I understand the complexity and nature of this project centered around Ids and configuration.

Not trying to nitpick, just elaborating on my initial thoghts. I'm okay with merging this PR since these points aren't blockers.

MiquelAdell commented:
Hi @MatiasArriola I've seen you made some comments and approved the PR. Should I merge it or wait for @9sneha-n to address the comments?

Thanks!

MatiasArriola commented:
> Hi @MatiasArriola I've seen you made some comments and approved the PR. Should I merge it or wait for @9sneha-n to address the comments?
> 
> Thanks!

@MiquelAdell If @9sneha-n agrees with my latest comments and we have some time, it would be great to refactor before merging. Otherwise, we can merge it as is.

MiquelAdell commented:
ok by me, I'll wait then. Thanks!

9sneha-n commented:
> @9sneha-n I was referring to the new `autoUpdateIndicationLinks` function more specifically. My reasoning is that the code in the hook primarily transforms domain entities `Questionnaire`, `QuestionnaireSection`, `Question` based on specific rules (e.g. linking certain questions bidirectionally) as a preparation for the save operation. The validation limiting a treatment to a maximum of 5 indications feels like a business rule rather than presentation logic. Another way to evaluate this is to ask: If we switched to a different presentation layer, would we need to rewrite this logic?
> 
> (unrelated to this PR) Ideally the data layer shouldn't be directly referenced by the presentation layer, though I understand the complexity and nature of this project centered around Ids and configuration.
> 
> I’m elaborating on my initial comment for clarity - these points aren’t blockers, and I’m okay with merging this PR if you consider it appropriate.

@MatiasArriola Done, addressed the review comments.

MatiasArriola reviewed (approved):
Thanks @9sneha-n!
